pkg/grpc/podstore/podstore.go

```diff
@@ -246,7 +246,7 @@ func (s store) MarkPodFailed(_ context.Context, req *podstore_protos.MarkPodFail
 	// we don't really need the CAS properties of MutateStatus but there
 	// should be only one system trying to write the status record so it
 	// doesn't hurt.
-	err = s.podStatusStore.MutateStatus(podUniqueKey, mutator)
+	err = s.podStatusStore.MutateStatus(ctx, podUniqueKey, mutator)
```

Comment:
I think this will always fail because `ctx` doesn't have the necessary metadata to perform a transaction. To see what i'm talking about look at `transaction.New()` which takes a context as input and derives a new context from the input one that has a value. The rest of the functions like `Add` will fail if the passed context doesn't have this value on it.

Probably I should have written it a bit differently so that the functions in the `transaction` package don't take the `context.Context` type but some different type like `transaction.Context` which is a superset of the `context.Context` interface but is guaranteed to have the data it needs. that way the compiler would complain at you here instead of just failing at run time.

All of that is a long winded way of saying that before this line you should do ```newCtx, cancel := transaction.New(ctx)
defer cancel()
```

and then pass `newCtx` to `MutateStatus`. Let me know if any of that doesn't make sense I just threw a lot at you